Client/Frontend/Browser/BrowserViewController/BrowserViewController+KeyCommands.swift

```diff
@@ -148,19 +161,91 @@ extension BrowserViewController {
         keyboardPressesHandler.reset()
     }
 
-    @objc private func showTabTrayKeyCommand() {
-        TelemetryWrapper.recordEvent(category: .action, method: .press, object: .keyCommand, extras: ["action": "show-tab-tray"])
-        showTabTray()
+    @objc private func selectFirstTab() {
+        selectTab(number: 0)
     }
 
-    @objc private func moveURLCompletionKeyCommand(sender: UIKeyCommand) {
-        guard let searchController = self.searchController else {
+    @objc private func selectTabTwo() {
+        selectTab(number: 1)
+    }
+
+    @objc private func selectTabThree() {
+        selectTab(number: 2)
+    }
+
+    @objc private func selectTabFour() {
+        selectTab(number: 3)
+    }
+
+    @objc private func selectTabFive() {
+        selectTab(number: 4)
+    }
+
+    @objc private func selectTabSix() {
+        selectTab(number: 5)
+    }
+
+    @objc private func selectTabSeven() {
+        selectTab(number: 6)
+    }
+
+    @objc private func selectTabEight() {
+        selectTab(number: 7)
+    }
+
+    @objc private func selectLastTab() {
+        guard let currentTab = tabManager.selectedTab else {
             return
         }
 
-        searchController.handleKeyCommands(sender: sender)
+        let tabs = currentTab.isPrivate ? tabManager.privateTabs : tabManager.normalTabs
+        selectTab(number: tabs.count - 1)
+    }
+
+    /// Select a certain tab number - If number is greater than the present number of tabs, select the last tab
+    /// - Parameter number: The 0 indexed tab number to select
+    private func selectTab(number: Int) {
+        guard let currentTab = tabManager.selectedTab else {
+            return
+        }
+
+        let tabs = currentTab.isPrivate ? tabManager.privateTabs : tabManager.normalTabs
+        // Do not continue if the index of the new tab to select is the current one
+        guard let currentTabIndex = tabs.firstIndex(of: currentTab), currentTabIndex != number else {
+            return
+        }
+
+        if tabs.count > number {
+            tabManager.selectTab(tabs[number])
+            keyboardPressesHandler.reset()
+        } else if let lastTab = tabs.last {
+            tabManager.selectTab(lastTab)
+            keyboardPressesHandler.reset()
+        }
+    }
+
+    // MARK: Zoom
+
+    @objc private func zoomIn() {
+        guard let currentTab = tabManager.selectedTab else { return }
```

Comment:
Possibly want to check that `firefoxHomeViewController == nil` here so you don't change the zoom of the tab while in overlay mode.